src/app/organization/loan-provisioning-criteria/edit-loan-provisioning-criteria/edit-loan-provisioning-criteria.component.ts

```diff
@@ -0,0 +1,221 @@
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { OrganizationService } from 'app/organization/organization.service';
+import { Router, ActivatedRoute } from '@angular/router';
+
+@Component({
+  selector: 'mifosx-edit-loan-provisioning-criteria',
+  templateUrl: './edit-loan-provisioning-criteria.component.html',
+  styleUrls: ['./edit-loan-provisioning-criteria.component.scss']
+})
+export class EditLoanProvisioningCriteriaComponent implements OnInit {
+
+   /** Criteria form. */
+   provisionCriteriaForm: FormGroup;
+   /** Product form. */
+   provisionProductForm: FormGroup;
+   /** Standard form. */
+   provisionStandardForm: FormGroup;
+   /** Sub Standard form. */
+   provisionSubStandardForm: FormGroup;
+   /** Doubtful form. */
+   provisionDoubtfulForm: FormGroup;
+   /** Loss form. */
+   provisionLossForm: FormGroup;
+   /** Is Linear. */
+   isLinear = false;
+   /** Provisioning Data. */
+   loanProvisioningTemplate: any;
+   /** Selected Products. */
+   selectedProducts: any = [];
+   /** Not Selected Products. */
+   refusalProducts: any;
+
+   /**
+    * Retrieves the offices data from `resolve`.
+    * @param {FormBuilder} formBuilder Form Builder.
+    * @param {OrganizationService} organizationService Organization Service.
+    * @param {ActivatedRoute} route Activated Route.
+    * @param {Router} router Router for navigation.
+    */
+
+  constructor(
+    private formBuilder: FormBuilder,
+    private organizationService: OrganizationService,
+    private router: Router,
+    private route: ActivatedRoute,
+  ) {
+    this.route.data.subscribe((data: { selectedLoanProvisioning: any }) => {
+      this.loanProvisioningTemplate = data.selectedLoanProvisioning;
+      this.loanProvisioningTemplate.definitions = this.loanProvisioningTemplate.definitions.sort((x: any, y: any) => {if (x.categoryId > y.categoryId) { return 1; }if (x.categoryId < y.categoryId) { return -1; }return 0; });
+    });
+  }
+
+  ngOnInit() {
+    this.loadProvisionForm();
+  }
+
+  /**
+   * Load the Provision Form.
+   */
+
+  loadProvisionForm() {
+    this.provisionCriteriaForm = this.formBuilder.group({
+      'provisioning_criteria': [this.loanProvisioningTemplate.criteriaName, Validators.required],
+    });
+
+    this.provisionProductForm = this.formBuilder.group({
+      'products': [this.loanProvisioningTemplate.selectedLoanProducts[0]],
+      'selectedProducts': ['']
+    });
+
```

Comment:
this code doesn't look 'dry', is it absolutely essential to have 4 separate forms? form fields are of each form are the same so cant a single form be re-used? I haven't studied this component in detail so it may be absolutely necessary to have 4 separate forms but if you can try to make it compact

Reply:
This was implemented using mat-stepper. Therefore separate forms were needed for the separate steppers. I tried with one form. But it was giving me an error in the console and did not work properly as I expected

Reply:
@abhi40308 can you comment upon the re-usability of this? I haven't used mat-stepper.

Reply:
> @abhi40308 can you comment upon the re-usability of this? I haven't used mat-stepper.

You can refer the following url if you want to get an idea about how it works.
https://material.angular.io/components/stepper/examples